parallel/client.py

```diff
@@ -192,11 +192,16 @@ def verify(self):
         """
         See abstract class for documentation.
         """
-        futures = [stub.Status.future(service_pb2.StatusRequest()) for stub in self.stubs]
-        workers_status = [x.result() for x in futures]
-        for field in ("nvidia_driver", "git_sha",):
-            # All fields should be the same
-            host_values = {self.hosts[i]: getattr(x, field) for i, x in enumerate(workers_status)}
-            uni_vals = set(host_values.values())
-            assert len(uni_vals) == 1, f"Not all hosts agreed for {field}: {host_values}"
-            assert all(uni_vals), f"Missing values for {field}: {host_values}"
+        prev_vals = {}
+        for host, stub in zip(self.hosts, self.stubs):
+            try:
+                status = stub.Status(service_pb2.StatusRequest())
```

Comment:
This has a default timeout of 20 seconds, could reduce it to 5/10, though might cause us some headache later.

Reply:
What are common reasons for timing out?

Reply:
- No connection/No server
- Slow connection <- This one might bite us if we set it too low